### PR TITLE
FSET-1966: Add ALLOCATION_EXPIRED state and Scheduled job

### PR DESCRIPTION
**Pull request by fayimora**

### CODE REVIEW

**Comment by fayimora** [comment]
Please squash when merging this. I messed up my commits 🙈

File: test/services/application/ApplicationServiceSpec.scala [hunk 1]
```diff
@@ -97,36 +98,69 @@ class ApplicationServiceSpec extends PlaySpec with BeforeAndAfterEach with Mocki
...
       verifyNoMoreInteractions(personalDetailsRepositoryMock, contactDetailsRepositoryMock, authProviderClientMock, auditServiceMock)
     }
   }
 
+  "process expired applications" should {
+    "set candidates who have not confirmed their booking to allocation expired and " +
+      "inform the candidate with an email" in new ApplicationServiceFixture {
+      when(appRepositoryMock.nextApplicationPendingAllocationExpiry).thenReturn(Future.successful(Some(expiringAllocation)))
+      when(authProviderClientMock.findByUserId(any[String])(any[HeaderCarrier])).thenReturn(Future.successful(Some(currentUser)))
+      when(appRepositoryMock.updateStatus(any[String], any[model.ApplicationStatuses.EnumVal])).thenReturn(Future.successful())
```
Comment (henricook) [inline test/services/application/ApplicationServiceSpec.scala]: There should be an explicit unit here (()) - or use thenReturnAsync with no parameters (from testkit.MockitoImplicits?)

File: app/services/testdata/AllocationExpiredStatusGenerator.scala [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    val newStatus = ApplicationStatuses.AllocationExpired
+
+    AllocationExpiredStatusGenerator.SlotFindingLockObj.synchronized {
+      for {
+        candidateInPreviousStatus <- previousStatusGenerator.generate(generationId, generatorConfig.copy(confirmedAllocation = false))
+        randomAssessment <- getApplicationAssessment(candidateInPreviousStatus)
+        _ <- aaRepository.create(List(randomAssessment))
+        expiryDate = randomAssessment.expireDate//.minusYears(2)
+        _ <- otRepository.saveCandidateAllocationStatus(candidateInPreviousStatus.applicationId.get, newStatus, Some(expiryDate))
+//        _ <- appRepository.updateAllocationExpiryDate(candidateInPreviousStatus.applicationId.get, expiryDate)
```
Comment (henricook) [inline app/services/testdata/AllocationExpiredStatusGenerator.scala]: Comment probably shouldn't still be here

Comment (fayimora) [reply app/services/testdata/AllocationExpiredStatusGenerator.scala]: Working on this locally to fix the TDG so I could remove it now but the next PR will get rid of it anyways. This PR just unblocks other tickets

**Pull request merged**